components/script/dom/idbrequest.rs

```diff
@@ -161,9 +161,9 @@ impl RequestListener {
                             Self::handle_async_request_error(&global, cx, request, e);
                             return;
                         };
-                        array.push(val.get());
+                        values.push(val.get());
```

Comment:
This adds the values onto the end of the already initialized vector. We either need to start with an empty vector or update the existing values in the initialized vector.